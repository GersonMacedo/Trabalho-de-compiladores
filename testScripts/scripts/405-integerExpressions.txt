! -s RUN --disable-logs printer builder
! should correctly calculate all the expressions
program p ;
    var i1 : integer;
    var i2 : integer;
begin
    print 1 + 2;
    print 5 - 3;
    print 2 - 4;
    print 3 * 4;
    print 12 / 3;
    print 14 / 3;
    print 15 / 3;

    i1 := 3;
    i2 := 4;
    print i1 + i2 * i2;
    print (i1 + i2) * i2;
    print i1 * i2 + i2;
    print i1 * (i2 + i2);

    print i1 - i2 * i2;
    print (i1 - i2) * i2;
    print i1 * i2 - i2;
    print i1 * (i2 - i2);

    print 4 + 4 / 2;
    print (4 + 4) / 2;
    print 4 / 4 + 2;
    print 4 / (4 + 2);

    print 4 - 4 / 2;
    print (4 - 4) / 2;
    print 4 / 4 - 2;
    print 4 / (4 - 2);

    print 3 + 4 - 10 - 2 + 7;
    print 4 * 5 / 10 * 3 / 2;
end .