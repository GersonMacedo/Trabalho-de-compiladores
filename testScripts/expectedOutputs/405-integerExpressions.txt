
Performing build generation and running at testScripts/scripts/405-integerExpressions.txt
No errors found
Building the program

      PUSH      2
      LOADL     1
      LOADL     2
      CALL      add
      CALL      putint
      CALL      puteol
      LOADL     5
      LOADL     3
      CALL      sub
      CALL      putint
      CALL      puteol
      LOADL     2
      LOADL     4
      CALL      sub
      CALL      putint
      CALL      puteol
      LOADL     3
      LOADL     4
      CALL      mult
      CALL      putint
      CALL      puteol
      LOADL     12
      LOADL     3
      CALL      div
      CALL      putint
      CALL      puteol
      LOADL     14
      LOADL     3
      CALL      div
      CALL      putint
      CALL      puteol
      LOADL     15
      LOADL     3
      CALL      div
      CALL      putint
      CALL      puteol
      LOADL     3
      STORE(1)  [0]SB      #i1
      LOADL     4
      STORE(1)  [1]SB      #i2
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      LOAD(1)   [1]SB      #i2
      CALL      mult
      CALL      add
      CALL      putint
      CALL      puteol
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      CALL      add
      LOAD(1)   [1]SB      #i2
      CALL      mult
      CALL      putint
      CALL      puteol
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      CALL      mult
      LOAD(1)   [1]SB      #i2
      CALL      add
      CALL      putint
      CALL      puteol
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      LOAD(1)   [1]SB      #i2
      CALL      add
      CALL      mult
      CALL      putint
      CALL      puteol
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      LOAD(1)   [1]SB      #i2
      CALL      mult
      CALL      sub
      CALL      putint
      CALL      puteol
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      CALL      sub
      LOAD(1)   [1]SB      #i2
      CALL      mult
      CALL      putint
      CALL      puteol
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      CALL      mult
      LOAD(1)   [1]SB      #i2
      CALL      sub
      CALL      putint
      CALL      puteol
      LOAD(1)   [0]SB      #i1
      LOAD(1)   [1]SB      #i2
      LOAD(1)   [1]SB      #i2
      CALL      sub
      CALL      mult
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      LOADL     2
      CALL      div
      CALL      add
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      CALL      add
      LOADL     2
      CALL      div
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      CALL      div
      LOADL     2
      CALL      add
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      LOADL     2
      CALL      add
      CALL      div
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      LOADL     2
      CALL      div
      CALL      sub
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      CALL      sub
      LOADL     2
      CALL      div
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      CALL      div
      LOADL     2
      CALL      sub
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     4
      LOADL     2
      CALL      sub
      CALL      div
      CALL      putint
      CALL      puteol
      LOADL     3
      LOADL     4
      CALL      add
      LOADL     10
      CALL      sub
      LOADL     2
      CALL      sub
      LOADL     7
      CALL      add
      CALL      putint
      CALL      puteol
      LOADL     4
      LOADL     5
      CALL      mult
      LOADL     10
      CALL      div
      LOADL     3
      CALL      mult
      LOADL     2
      CALL      div
      CALL      putint
      CALL      puteol
      POP(0)    2
      HALT
Builder::Writing object to obj.tam
********** TAM Interpreter (Java Version 2.1) **********
3
2
-2
12
4
4
5
19
28
16
24
-13
-4
8
0
6
4
3
0
2
0
-1
2
2
3

Program has halted normally.
